server_tests/test/clientTests.js

```diff
@@ -1307,63 +1206,59 @@ describe("RUST API TEST SUITE", function() {
             }
 
             this.timeout(timeoutValue);
-            testClient.startSession()
+            return testClient.startSession()
             .then( function(res) {
                 sessionId = res.body.results.sessionId;
-                done();
             });
         });
 
-        beforeEach( function(done) {
+        beforeEach( function() {
             this.timeout(timeoutValue);
             Utility.restoreDefaultOptions(defaultOptionsFile)
             .then(function() {
                 Utility.deleteAllImages()
                 .then(function() {
-                    done();
                 });
             });
         });
 
-        after( function(done) {
-            testClient.closeSession(sessionId)
+        after( function() {
+            return testClient.closeSession(sessionId)
             .then( function() {
-                done();
             });
         });
 
-        it('Expect missingParameter Error. sessionId is mandatory for command camera._bublTimelapse', function(done) {
-            testClient.bublTimelapse()
-            .then( function(res) {
-                var err = Comparison.missingParameterError(res);
-                done(err);
+        it('Expect missingParameter Error. sessionId is mandatory for command camera._bublTimelapse', function() {
+            return testClient.bublTimelapse()
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.missingParameterError(err);
             });
         });
 
-        it('Expect invalidParameterValue Error. camera._bublTimelapse expects active session\'s sessionId', function(done) {
-            testClient.bublTimelapse(sessionId + '0')
-            .then( function(res) {
-                var err = Comparison.invalidParameterValueError(res);
-                done(err);
+        it('Expect invalidParameterValue Error. camera._bublTimelapse expects active session\'s sessionId', function() {
+            return testClient.bublTimelapse(sessionId + '0')
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.invalidParameterValueError(err);
             });
         });
 
-        it('Expect cameraInExclusiveUse Error. camera._bublTimelapse cannot be run when another timelapse capture procedure is already active', function(done) {
+        it('Expect cameraInExclusiveUse Error. camera._bublTimelapse cannot be run when another timelapse capture procedure is already active', function() {
             this.timeout(timeoutValue * 2);
             var firstTimelapseDone = false;
             var startTime = Date.now();
             var timeSendAnother = 1000;
             var stopTime = 5000;
             var stopSent = false;
             var stopErr;
+
             //instantiate the first camera._bublTimelapse command
-            testClient.bublTimelapse(sessionId, function(res) {
+            return testClient.bublTimelapse(sessionId, function(res) {
```

Comment:
None of the code after this return actually gets executed. The try after is weird and still uses the old `done()`, needs deleting. There are two promises here, so again, needs deferred or similar.